src/actions/ACMActivator.ts

```diff
@@ -190,20 +190,37 @@ export class ACMActivator implements IExecutor {
       } else {
         this.logger.debug(`Device ${clientObj.uuid} activated in admin mode.`)
         clientObj.ciraconfig.status = 'activated in admin mode.'
+        clientObj.activationStatus = true
         this.clientManager.setClientObject(clientObj)
 
-        /* Set MEBx password called after the activation as the API is accessible only with admin user */
-        await this.setMEBxPassword(clientId, clientObj)
+        await this.waitAfterActivation(clientId, clientObj, async () => {
+          /* Set MEBx password called after the activation as the API is accessible only with admin user */
+          await this.setMEBxPassword(clientId, clientObj)
 
-        /* After the activation, control goes to Network configuration */
-        clientObj.action = ClientAction.NETWORKCONFIG
-        await this.networkConfigurator.execute(message, clientId)
+          /* After the activation, control goes to Network configuration */
+          // clientObj.action = ClientAction.NETWORKCONFIG
+          await this.networkConfigurator.execute(message, clientId)
+        })
       }
     } else {
       throw new RPSError(`Device ${clientObj.uuid} sent an invalid response.`)
     }
   }
 
+  async waitAfterActivation (clientId: string, clientObj: ClientObject, callback): Promise<void> {
+    this.logger.info(`waiting for ${EnvReader.GlobalEnvConfig.delayTimer} seconds after activation`)
+    const endTime: Date = new Date()
+    endTime.setSeconds(endTime.getSeconds() + EnvReader.GlobalEnvConfig.delayTimer)
+    const intervalID = setInterval(async () => {
+      await this.amtwsman.batchEnum(clientId, '*AMT_GeneralSettings')
+      const dt = new Date()
+      if (dt >= endTime) {
+        clearInterval(intervalID)
+        callback()
```

Comment:
Lets use a promise instead of a callback to avoid nested code

Reply:
Removed